libnavigation-core/src/main/java/com/mapbox/navigation/core/directions/session/RoutesObserver.kt

```diff
@@ -30,10 +30,12 @@ fun interface RoutesObserver {
  * @param navigationRoutes list of currently maintained routes
  * @param ignoredRoutes list of alternative routes that were ignored
  *  because they are invalid for navigation. See [IgnoredRoute] for details.
+ * @param initialLegIndex the initial leg index of the primary route.
  * @param reason why the routes have been updated (re-route, refresh route, and etc.)
  */
 class RoutesUpdatedResult internal constructor(
     val navigationRoutes: List<NavigationRoute>,
+    val initialLegIndex: Int,
```

Comment:
How is the SDK user supposed to calculate initial leg index for the case when user switches between alternatives clicking on them?

Reply:
Depending on how the new routes are set. They are probably set by the user. They do sth like:
```
val routes = listOf(newPrimaryRoute + mapboxNavigation.getNavigationRoutes().remove(newPrimaryRoute))
mapboxNavigation.setNavigationRoutes(routes)
```
Here we can only tell the user they should pass the leg index, we can't really do anything from our end.
All we see is new routes incoming with initialLegIndex = 0. Changing that even if we had enough data would be overriding user input, which just doesn't sound right.

Reply:
what do you think about exposing following API:

For the case when user manually switches to an alternative route by clicking on it: 
```
mapboxNavigation.switchToAlternative(alternativeId)
```
and for reroute case:
```
mapboxNavigation.deviateToAlternative(alternativeId)
```
?

Reply:
We can, but there's absolutely no guarantee that customers will switch to it.